docs/cloud/pouta/tutorials/image-pipeline.md

```diff
@@ -0,0 +1,650 @@
+
+For the sake of this tutorial, the transformed image simply corresponds to the input image whose colors have been inverted, sometimes also called "reversed".
+
+## Step 1: creating buckets in Allas
+
+We open a new terminal window, to which we will refer using the name `terminal_A`.
+We use `terminal_A` for all the commands dealing with Allas.
+
+To create buckets in Allas, we need to have a working command-line interface for it.
+If we haven't set up such interface before on our workstation, we follow the [instructions on how to install and configure s3cmd](../../../data/Allas/using_allas/s3_client.md#getting-started-with-s3cmd).
+
+We can test the correct functioning of the command-line interface by simply listing all the buckets currently in our project.
+An example of the command and its expected output follows:
+```
+$ s3cmd ls
+2021-07-14 15:14  s3://bucket1
+2020-01-14 17:40  s3://bucket2
+...
+```
+Please note that the list can also be empty, if we haven't created any bucket in our project before.
+
+We create the input and the output buckets for our pipeline using the following commands:
+```
+$ s3cmd mb s3://input_bucket
+Bucket 's3://input_bucket/' created
+$ s3cmd mb s3://output_bucket
+Bucket 's3://output_bucket/' created
+```
+
+!!! warning
+    Bucket names must be unique.
+    If another user has already selected the same name, bucket creation command will fail:
+    ```
+    $ s3cmd mb s3://input_bucket
+    ERROR: Bucket 'input_bucket' already exists
+    ERROR: S3 error: 409 (BucketAlreadyExists)
+    ```
+    In such case, we can just select a different name and retry the command.
+
+In the rest of the tutorial, we assume:
+
+* the name of the bucket used for uploading images to the pipeline is `input_bucket`
+* the name of the bucket used by the pipeline to make available its outputs is `output_bucket`
+
+## Step 2: creating database in Pukki
+
+We open a second terminal window, to which we will refer using the name `terminal_B`.
+We use `terminal_B` for all the commands dealing with Pukki.
+
+Having a working command-line interface for Pukki is a prerequisite for continuing with the tutorial.
+If we haven't set it up before, we follow the [instructions on how to install and configure Pukki command-line interface](../../dbaas/cli.md#getting-started).
+
+We can test the correct functioning of the command-line interface by simply listing the available types of database.
+An example of the command and its expected output follows:
+```
+$ openstack datastore list
++--------------------------------------+------------+
+| ID                                   | Name       |
++--------------------------------------+------------+
+| 71920375-6967-466e-b955-8ee8629312b7 | postgresql |
+| 1a8efda2-7bb7-4c52-9eab-e251fd18323c | mariadb    |
++--------------------------------------+------------+
+```
+
+We now create the database that we use to log the actions of the pipeline by issuing the following command:
```

Comment:
```suggestion
We now create the database that we will use to log the actions of the pipeline by issuing the following command:
```

Added the future tense, otherwise the sentence sounded weird.